File: backend/python-service/alphaflow-core/src/alphaflow/market_data/keys.py
def indicator_key(exchange: str, market: str, symbol: str, interval: str) -> str:
    return f"{exchange_code(exchange)}:{market}:ind:{symbol}:{interval}"


def indicator_window_key(exchange: str, market: str, symbol: str, interval: str) -> str:
    return f"{exchange_code(exchange)}:{market}:indwin:{symbol}:{interval}"


def indicator_realtime_key(exchange: str, market: str, symbol: str, interval: str) -> str:
    return f"{exchange_code(exchange)}:{market}:indrt:{symbol}:{interval}"


def data_health_key(exchange: str, market: str, symbol: str, interval: str) -> str:
    return f"{exchange_code(exchange)}:{market}:health:{symbol}:{interval}"


def last_price_key(exchange: str, market: str, symbol: str) -> str:
    return f"{exchange_code(exchange)}:{market}:lp:{symbol}"


def mark_price_key(exchange: str, market: str, symbol: str) -> str:
    return f"{exchange_code(exchange)}:{market}:mp:{symbol}"


def kline_base_key(exchange: str, market: str, symbol: str, interval: str) -> str:
    return f"{exchange_code(exchange)}:{market}:k:{symbol}:{interval}"


def kline_index_key(exchange: str, market: str, symbol: str, interval: str) -> str:
    return kline_base_key(exchange, market, symbol, interval) + ":idx"


def kline_data_key(exchange: str, market: str, symbol: str, interval: str) -> str:
    return kline_base_key(exchange, market, symbol, interval) + ":data"


def exchange_code(exchange: str) -> str:
    if exchange == "binance":
        return "bn"
    return exchange

File: backend/go-service/pkg/marketkeys/keys.go
package marketkeys

import "fmt"

func KlineKey(exchange string, market string, symbol string, interval string) string {
	return fmt.Sprintf("%s:%s:k:%s:%s", ExchangeCode(exchange), market, symbol, interval)
}

func IndicatorKey(exchange string, market string, symbol string, interval string) string {
	return fmt.Sprintf("%s:%s:ind:%s:%s", ExchangeCode(exchange), market, symbol, interval)
}

func IndicatorLastKey(exchange string, market string, symbol string, interval string) string {
	return fmt.Sprintf("%s:%s:ind:last:%s:%s", ExchangeCode(exchange), market, symbol, interval)
}

func IndicatorWindowKey(exchange string, market string, symbol string, interval string) string {
	return fmt.Sprintf("%s:%s:indwin:%s:%s", ExchangeCode(exchange), market, symbol, interval)
}

func IndicatorWindowLatestKey(exchange string, market string, symbol string, interval string) string {
	return fmt.Sprintf("%s:%s:indwin:latest:%s:%s", ExchangeCode(exchange), market, symbol, interval)
}

func IndicatorWindowLastKey(exchange string, market string, symbol string, interval string) string {
	return fmt.Sprintf("%s:%s:indwin:last:%s:%s", ExchangeCode(exchange), market, symbol, interval)
}

func IndicatorRealtimeKey(exchange string, market string, symbol string, interval string) string {
	return fmt.Sprintf("%s:%s:indrt:%s:%s", ExchangeCode(exchange), market, symbol, interval)
}

func LastPriceKey(exchange string, market string, symbol string) string {
	return fmt.Sprintf("%s:%s:lp:%s", ExchangeCode(exchange), market, symbol)
}

func MarkPriceKey(exchange string, market string, symbol string) string {
	return fmt.Sprintf("%s:%s:mp:%s", ExchangeCode(exchange), market, symbol)
}

func ExchangeCode(exchange string) string {
	switch exchange {
	case "binance":
		return "bn"
	default:
		return exchange
	}
}
